src/main/java/com/google/gwt/dom/client/DOMImpl.java

```diff
@@ -19,15 +19,108 @@
 
 import com.google.common.base.MoreObjects;
 import com.google.gwt.core.client.JsArray;
+import com.google.gwt.event.dom.client.BlurEvent;
+import com.google.gwt.event.dom.client.ChangeEvent;
 import com.google.gwt.event.dom.client.ClickEvent;
+import com.google.gwt.event.dom.client.FocusEvent;
+import com.google.gwt.event.dom.client.DomEvent.Type;
+import com.google.gwt.user.client.DOM;
+import com.google.gwt.user.client.Event;
+import com.google.gwt.user.client.EventListener;
 
 public class DOMImpl {
 	
 	static final DOMImpl impl = new DOMImpl();
 	
-	public void buttonClick(ButtonElement button) {
-		button.fireEvent(ClickEvent.getType());
-	}
+	// Element In Focus (some browsers actually have working implementation document.activeElement)
+	private Element activeElement;
+	
+    private void switchFocusTo(Element element) {
+        if (activeElement == element) {
+            return;
+        } else {
+            if (activeElement != null && isAttached(activeElement)) {
+                if (activeElement instanceof InputElement) {
+                    fireValueChangeIfAny((InputElement)activeElement);
+                }
+                activeElement.fireEvent(BlurEvent.getType());
+            }
+            element.fireEvent(FocusEvent.getType());
+            activeElement = element;
+        }
+    }
+
+    private boolean isAttached(Element elem) {
+        return DOM.getEventListener(elem) != null;
+    }
+
+    private void fireValueChangeIfAny(InputElement element) {
```

Comment:
DOMImpl.buttonClick is part of the original GWT source, and it's used by ButtonElement.click(), that's why it has to be implemented.

I think it's a good approach to keep static helpers separately. It's the single responsibility principle. You don't bundle things up to a big monolith. Other developers might need different helpers or other users would want a different API for their helpers. You can start a gwt-mock-helpers project, that hosts the helpers we commonly use, and then others can use that too. But these helpers are definitiely not part of gwt-mock.